Objects/frameobject.c

```diff
@@ -832,46 +826,60 @@ frame_alloc(PyCodeObject *code)
         --state->numfree;
         f = state->free_list;
         state->free_list = state->free_list->f_back;
-        if (Py_SIZE(f) < extras) {
-            PyFrameObject *new_f = PyObject_GC_Resize(PyFrameObject, f, extras);
-            if (new_f == NULL) {
-                PyObject_GC_Del(f);
-                return NULL;
-            }
-            f = new_f;
-        }
         _Py_NewReference((PyObject *)f);
     }
-
-    extras = code->co_nlocals + ncells + nfrees;
-    f->f_valuestack = f->f_localsplus + extras;
-    for (Py_ssize_t i=0; i < extras; i++) {
-        f->f_localsplus[i] = NULL;
-    }
+    f->f_localsptr = localsarray;
+    f->f_own_locals_memory = owns;
     return f;
 }
 
+int
+_PyFrame_TakeLocals(PyFrameObject *f)
+{
+    assert(f->f_own_locals_memory == 0);
+    assert(f->f_stackdepth == 0);
+    int size = frame_nslots(f);
```

Comment:
@markshannon, this gives a compiler error because frame_nslots() returns a Py_ssize_t.